### PR TITLE
[NN] Dot-based GAT Convolution Module.

### PR DESCRIPTION
**Pull request by zhjwy9343**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [X] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [X] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/nn/pytorch/conv/dotgatconv.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+        .. math::
+            \alpha_{i, j} = \mathrm{softmax_i}(e_{ij}^{l})
+
+            e_{ij}^{l} = ({W_i^{(l)} h_i^{(l)}})^T \cdot {W_j^{(l)} h_j^{(l)}}
+
+        where :math:`W_i` and :math:`W_j` transform node :math:`i`'s and node :math:`j`'s
+        features into the same dimension, so that when compute note features' similarity,
+        we can use dot-product.
+    """
```
Comment (yzh119) [inline python/dgl/nn/pytorch/conv/dotgatconv.py:26]: Please introduce `__init__` function parameters in the docstring by
```
Parameters
----------------
in_feats : int
    your description
out_feats : int
    your description
```

File: python/dgl/nn/pytorch/conv/dotgatconv.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                 ):
+        super(DotGatConv, self).__init__()
+        self._in_src_feats, self._in_dst_feats = expand_as_pair(in_feats)
+        self._out_feats = out_feats
+
+        if isinstance(in_feats, tuple):
+            self.fc_src = nn.Linear(self._in_src_feats, self._out_feats, bias=False)
+            self.fc_dst = nn.Linear(self._in_dst_feats, self._out_feats, bias=False)
+        else:
+            self.fc = nn.Linear(self._in_src_feats, self._out_feats, bias=False)
```
Comment (yzh119) [inline python/dgl/nn/pytorch/conv/dotgatconv.py:40]: How are `self.fc_src`, `self.fc_dst` and `self.fc` initialized?
Describe them in 
```
def reset_parameters():
    # TODO
```

File: python/dgl/nn/pytorch/conv/dotgatconv.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        where :math:`\alpha_{ij}` is the attention score bewteen node :math:`i` and node :math:`j`:
+
+        .. math::
+            \alpha_{i, j} = \mathrm{softmax_i}(e_{ij}^{l})
+
+            e_{ij}^{l} = ({W_i^{(l)} h_i^{(l)}})^T \cdot {W_j^{(l)} h_j^{(l)}}
+
+        where :math:`W_i` and :math:`W_j` transform node :math:`i`'s and node :math:`j`'s
+        features into the same dimension, so that when compute note features' similarity,
+        we can use dot-product.
```
Comment (yzh119) [inline python/dgl/nn/pytorch/conv/dotgatconv.py:25]: Unindent these lines.

File: python/dgl/nn/pytorch/conv/dotgatconv.py (hunk 2 above)
Comment (zhjwy9343) [reply python/dgl/nn/pytorch/conv/dotgatconv.py:40]: This module uses PyTorch Linear, which has its default initialization method. So there is no need of initialize. Should we give this 'reset_parameters()' method as a requirement?

File: python/dgl/nn/pytorch/conv/dotgatconv.py (hunk 3 above)
Comment (zhjwy9343) [reply python/dgl/nn/pytorch/conv/dotgatconv.py:25]: Added these doc strings about input features

File: python/dgl/nn/pytorch/conv/dotgatconv.py (hunk 1 above)
Comment (zhjwy9343) [reply python/dgl/nn/pytorch/conv/dotgatconv.py:26]: Added doc strings about them.

**Pull request closed without merging**